.github/bin/free-disk-space.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -euo pipefail
+
+function free_up_disk_space_ubuntu()
+{
+    local pkgs='azure-cli aspnetcore-* dotnet-* ghc-* firefox*
+        google-chrome-stable google-cloud-cli libmono-* llvm-*
+        microsoft-edge-stable mono-* msbuild mysql-server-core-*
+        php-* php7* powershell*'
+    # Use apt patterns to only select real packages that match the names
+    # in the list above.
+    local pkgs=$(echo $pkgs | sed 's/[^ ]* */~n&/g')
```

Comment:
what does `~n` do?

nit: `local` is redundant (since `pkgs` is already defined in this scope).
also, `local` is a statement that always succeeds, so it masks any errors 
from the `=....` side. for this reason, `local` and non-const assignment should be separate lines

Reply:
`apt-get remove '~n ^dotnet'` removes all installed packages starting with dotnet